packages/gpl-core/src/badge.ts

```diff
@@ -0,0 +1,314 @@
+import { SDK } from ".";
+import { gql } from "graphql-request";
+import * as anchor from "@project-serum/anchor";
+import randomBytes from "randombytes";
+
+export interface GraphQLBadge {
+  address: string;
+  issuer: string;
+  holder: string;
+  update_authority: string;
+  schema: string;
+  metadata_uri: string;
+  refreshed_at?: Date;
+  slot_created_at?: Date;
+  slot_updated_at?: Date;
+  created_at?: Date;
+}
+
+export interface GraphQLIssuer {
+  address: string;
+  authority: string;
+  verified: boolean;
+  refreshed_at?: Date;
+  slot_created_at?: Date;
+  slot_updated_at?: Date;
+  created_at?: Date;
+}
+
+export class Badge {
+  private readonly sdk: SDK;
+
+  constructor(sdk: SDK) {
+    this.sdk = sdk;
+  }
+
+  public async getBadge(userAccount: anchor.web3.PublicKey) {
+    return await this.sdk.program.account.badge.fetch(userAccount);
+  }
+
+  public async getOrCreateBadge(
```

Comment:
I am not sure if we want to expose `getOrCreateBadge`, I think it is better to be explicit about whether someone wants to fetch a badge or create one.

Also, only one badge of a type can be issued to a profile.